src/solver/optimisation/post_process_commands.cpp

```diff
@@ -112,53 +112,93 @@ void RemixHydroPostProcessCmd::execute(const optRuntimeData& opt_runtime_data)
+    }
+}
+
 // -----------------------------
 //  DTG margin for adq patch
 // -----------------------------
-using namespace Antares::Data::AdequacyPatch;
-
-DTGmarginForAdqPatchPostProcessCmd::DTGmarginForAdqPatchPostProcessCmd(
-  const AdqPatchParams& adqPatchParams,
+DTGnettingAfterCSRcmd::DTGnettingAfterCSRcmd(
   PROBLEME_HEBDO* problemeHebdo,
   AreaList& areas,
   unsigned int thread_number) :
  basePostProcessCommand(problemeHebdo),
- adqPatchParams_(adqPatchParams),
  area_list_(areas),
  thread_number_(thread_number)
 {
 }
 
-/*!
-** Calculate Dispatchable margin for all areas after CSR optimization and adjust ENS
-** values if neccessary. If LOLD=1, Sets MRG COST to the max value (unsupplied energy cost)
-** */
-void DTGmarginForAdqPatchPostProcessCmd::execute(const optRuntimeData&)
+void DTGnettingAfterCSRcmd::execute(const optRuntimeData&)
 {
     for (uint32_t Area = 0; Area < problemeHebdo_->NombreDePays; Area++)
     {
         if (problemeHebdo_->adequacyPatchRuntimeData->areaMode[Area] != physicalAreaInsideAdqPatch)
             continue;
 
+        auto& hourlyResults = problemeHebdo_->ResultatsHoraires[Area];
+        const auto& scratchpad = area_list_[Area]->scratchpad[thread_number_];
+        const double unsuppliedEnergyCost = area_list_[Area]->thermal.unsuppliedEnergyCost;
+
         for (uint hour = 0; hour < nbHoursInWeek; hour++)
         {
-            auto& hourlyResults = problemeHebdo_->ResultatsHoraires[Area];
-            const auto& scratchpad = area_list_[Area]->scratchpad[thread_number_];
+            const bool isHourTriggeredByCsr = problemeHebdo_->adequacyPatchRuntimeData
+                    ->wasCSRTriggeredAtAreaHour(Area, hour);
+
             const double dtgMrg = scratchpad.dispatchableGenerationMargin[hour];
             const double ens = hourlyResults.ValeursHorairesDeDefaillancePositive[hour];
-            const bool triggered
-              = problemeHebdo_->adequacyPatchRuntimeData->wasCSRTriggeredAtAreaHour(Area, hour);
-            hourlyResults.ValeursHorairesDtgMrgCsr[hour] = recomputeDTG_MRG(triggered, dtgMrg, ens);
-            hourlyResults.ValeursHorairesDeDefaillancePositiveCSR[hour]
-              = recomputeENS_MRG(triggered, dtgMrg, ens);
 
-            if (triggered)
+            // Default value (when the hour is not triggered by CSR)
+            hourlyResults.ValeursHorairesDtgMrgCsr[hour] = dtgMrg;
+            hourlyResults.ValeursHorairesDeDefaillancePositiveCSR[hour] = ens;
+            // Marginal cost from solver is negative, and we need it positive, hence the '-' sign.
+            hourlyResults.MarginalCostsCSR[hour] = -hourlyResults.CoutsMarginauxHoraires[hour];
+
+            if (isHourTriggeredByCsr)
             {
-                const double unsuppliedEnergyCost = area_list_[Area]->thermal.unsuppliedEnergyCost;
-                hourlyResults.CoutsMarginauxHoraires[hour]
-                  = recomputeMRGPrice(hourlyResults.ValeursHorairesDtgMrgCsr[hour],
```

Comment:
About previous **recomputeMRGPrice(....)** : 
Note that first argument here is **Dtg Mrg Csr**.
But in the **recomputeMRGPrice(....)**'s definition, first argument is **ens Csr**, which was very misleading.
Now we no longer have this problem.